pandas/core/strings.py

```diff
@@ -2048,12 +2112,13 @@ def _get_series_list(self, others, ignore_index=False):
                     warnings.warn('list-likes other than Series, Index, or '
                                   'np.ndarray WITHIN another list-like are '
                                   'deprecated and will be removed in a future '
-                                  'version.', FutureWarning, stacklevel=3)
+                                  'version.', FutureWarning, stacklevel=4)
                 return (los, join_warn)
             elif all(not is_list_like(x) for x in others):
                 return ([Series(others, index=idx)], False)
         raise TypeError(err_msg)
 
+    @forbid_nonstring_types(['bytes', 'mixed', 'mixed-integer'])
```

Comment:
would it make sense rather than have a forbid decorator have an include decorator that accepts string, unicode? (e.g. wouldn't this happen to accept object dtypes that are non-strings but not in the included that you have written)?

Reply:
> @jreback: e.g. wouldn't this happen to accept object dtypes that are non-strings but not in the included that you have written

You're right, but this would be a very hard breaking change that I do not want to author like that. I responded similarly [here](https://github.com/pandas-dev/pandas/pull/23167#discussion_r225793121):
___________________________________

Ok in principle, but it's definitely necessary to allow `'mixed'`:
```
>>> lib.infer_dtype(pd.Series(['a', np.nan]))
'mixed'
```

Disallowing `'mixed-integer'` would also break existing code, but I mentioned in the OP:
> Finally, it would be possible to partially close #13877 by also excluding `'mixed-integer'` from the allowed inferred types.

Check out the example in #13877 - currently the `.str`-methods are generally permissive about just returning `NaN` if they can't compute individual values. Arguably, that's also the most desirable behaviour for things like `.str.startswith`. Could introduce an `'errors'`-kwarg where necessary, but all of that needs to happen on method level - the `.str`-constructor must allow these to pass.

Reply:
You also mentioned somewhere further up that the removal of the capabilities for bytes means that we should add a `.bytes` accessor (which was why I didn't try to push this PR through before 0.24).